openwisp_controller/config/tests/test_admin.py

```diff
@@ -614,6 +614,24 @@ def test_download_template_config(self):
         response = self.client.get(path)
         self.assertEqual(response.get('content-type'), 'application/octet-stream')
 
+    def test_template_has_download_config(self):
+        t = Template.objects.first()
+        path = reverse(f'admin:{self.app_label}_template_change', args=[t.pk])
+        r = self.client.get(path)
+        self.assertContains(r, 'Download configuration')
+
+    def test_template_not_have_download_config(self):
+        model_kwargs = {
+            'name': 'test-template',
+            'backend': 'netjsonconfig.OpenWrt',
+        }
+        t = Template(**model_kwargs)
+        t.full_clean()
+        t.save()
+        path = reverse(f'admin:{self.app_label}_template_change', args=[t.pk])
+        r = self.client.get(path)
+        self.assertNotContains(r, 'Download configuration')
```

Comment:
Thank you for bringing this up, I'm not sure why creating an empty template is even possible. I opened a bug report for this: https://github.com/openwisp/openwisp-controller/issues/276

So, this test is checking a case which looks like a bug rather than an intended feature, for this reason we should not add it.